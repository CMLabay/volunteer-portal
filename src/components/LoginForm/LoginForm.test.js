import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm'

describe('Login form component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <LoginForm></LoginForm>
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders teh UI as expected', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                <LoginForm></LoginForm>
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

})