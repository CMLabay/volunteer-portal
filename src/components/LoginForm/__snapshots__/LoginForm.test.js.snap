// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component renders teh UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      Email
    </label>
    <input
      className="Input form-input"
      id="LoginForm__user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      className="Input form-input"
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button "
    type="submit"
  >
    Login
  </button>
</form>
`;
